### PR TITLE
refactor(treesitter): rename query previewer to query editor

### PR DESCRIPTION
**Pull request by MariaSolOs**
As agreed in https://github.com/neovim/neovim/pull/24703#discussion_r1321719133.

### CODE REVIEW

**Comment by clason** [comment]
Maybe add a note to the doc string about the current limitations (only a "scratch query" for now)?

File: runtime/doc/news.txt [hunk 1]
```diff
@@ -135,7 +135,7 @@ The following new APIs and features were added.
     `vim.treesitter.language.register`.
   • The `#set!` directive now supports `injection.self` and `injection.parent` for injecting either the current node's language
     or the parent LanguageTree's language, respectively.
-  • Added `vim.treesitter.preview_query()`, for live editing of treesitter
+  • Added `vim.treesitter.query.edit()`, for live editing of treesitter
```
Comment (gpanders) [inline runtime/doc/news.txt:138]: This says `vim.treesitter.query.edit()` but the rest of the diff shows `vim.treesitter.edit_query()`? I think we settled on the former.

Comment (MariaSolOs) [reply runtime/doc/news.txt:138]: Ahhh my bad, got confused. Fixing it.

**Comment by clason** [comment]
Also, I just noticed a limitation of the current query editor: it does not play nicely with injected languages -- for example, you cannot test `markdown_inline` queries on a Markdown buffer.

Is it possible to add an optional argument to override the language for which the query editor is opened?

**Comment by clason** [comment]
> We can configure the command to accept a filetype argument though,

I'd say just specify the parser; no need for filetype lookups. And no need to add that to the command; an advanced feature like that is fine to reserve for the Lua function.

>  but that would be a different PR ;)

Sure :)

**Review by justinmk — APPROVED** [review]
Thanks for following up on this!

**Pull request merged**